Trustroots/trustroots
# Switch from bower to webpack

#### Proposed Changes

* switching from bower to webpack - [bower](https://bower.io/) says: _...psst! While Bower is maintained, we recommend using Yarn and Webpack or Parcel for front-end projects read how to migrate!_
* also prep for moving away from angular1 to react (see https://github.com/Trustroots/trustroots/issues/779#issuecomment-433696445 and https://github.com/Trustroots/trustroots/issues/638#issuecomment-415192512)

#### Testing Instructions

* run dev setup as normal
* ... which should run `npm install` and `npm run build` as part of it, if not, run those
* visit the site
* you should not notice anything changed (my favourite kind of contribution)

#### TODO

- [x] put bower in fire by integrating / rebasing with https://github.com/Trustroots/trustroots/pull/783
- [x] make tests work
- [x] build service worker scripts via webpack too
- [x] production build setup
- [x] check/fix profile networks (`Error: dst is undefined`)
- [x] check it works in docker
- [x] merge/adapt/rewrite dev scripts to use webpack build (and remove `build:webpack` npm script)

Some time later, or sooner, depending on motivation, but not essential for now, probably best in another PR:
- [ ] put gulp in fire
- [ ] style (css/less) via webpack
- [ ] add hot (module) reloading
- [ ] maybe use some cooler webpack options (splitting, etc) to make it suitable for production deployment
- [ ] reimplement `uib-templates.js` / `templates.js` building via webpack loader
- [ ] consider extracting and abstracting dependencies for legacy libs ([example from fs.de](https://gist.github.com/nicksellen/569a54762e9c2343cf27ffe15682a1aa))
- [ ] move away from massive lists of files in `config/assets/*`
- [ ] actually use js modules (currently all the actual `modules/*` files are written as if they are global) - can probably use some [webpack contexts](https://webpack.js.org/guides/dependency-management/) for globbing
- [ ] add es6 loader to webpack
- [ ] add at least 300 webpack plugins so computer can be used as heater in the coming winter
